src/test/java/ru/clevertec/product/service/impl/ProductServiceImplTest.java

```diff
@@ -0,0 +1,241 @@
+package ru.clevertec.product.service.impl;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import ru.clevertec.product.data.InfoProductDto;
+import ru.clevertec.product.data.ProductDto;
+import ru.clevertec.product.entity.Product;
+import ru.clevertec.product.mapper.ProductMapper;
+import ru.clevertec.product.repository.ProductRepository;
+import ru.clevertec.product.builders.InMemoryProductRepositoryTestData;
+import ru.clevertec.product.builders.ProductTestData;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class ProductServiceImplTest {
+
+    private static final UUID PRODUCT_UUID = UUID.fromString("a28fed6a-f58b-47d2-bcb6-4d89c77ea0ed");
+
+    @Mock
+    ProductRepository repositoryMock;
+
+    @Mock
+    ProductMapper mapperMock;
+
+    @InjectMocks
+    ProductServiceImpl productService;
+
+    @Captor
+    ArgumentCaptor<Product> productCaptor;
+
+    @Captor
+    ArgumentCaptor<UUID> uuidCaptor;
+
+    @Test
+    void getShouldReturnInfoProductDto() {
+        // given
+        InfoProductDto expected = ProductTestData.builder()
+                .withUuid(PRODUCT_UUID)
+                .build().createInfoProductDto();
+
+        Product product = ProductTestData.builder()
+                .withUuid(PRODUCT_UUID)
+                .build().createProduct();
+
+        when(repositoryMock.findById(PRODUCT_UUID)).thenReturn(Optional.of(product));
```

Comment:
<pre>
when(repositoryMock.findById(PRODUCT_UUID))
    .thenReturn(Optional.of(product));
</pre>